### ISSUE TITLE
Provide More Details on "Common Services"

### ISSUE THREAD

**Issue by taylorchasewhite** [dotnet/AspNetCore.Docs#22122]
Two requests in this:

1. For anyone moving to prerendering, they need to know what kind of services need to be registered over on the Server, as opposed to the client. Saying "Common Services" is not at all enlightening nor helpful top that end. Can you explain what kinds of services need to be placed here, vs. which should only be registered on the client?
2. Can we reference this article from the one discussing converting to PreRendering?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 53f3209c-2414-9b35-6245-383c9b7c6338
* Version Independent ID: cc8dfcd6-8d93-5aa3-c381-2b613e88cdbe
* Content: [ASP.NET Core Blazor WebAssembly additional security scenarios](https://docs.microsoft.com/en-us/aspnet/core/blazor/security/webassembly/additional-scenarios?view=aspnetcore-5.0)
* Content Source: [aspnetcore/blazor/security/webassembly/additional-scenarios.md](https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/blazor/security/webassembly/additional-scenarios.md)
* Product: **aspnet-core**
* Technology: **aspnetcore-blazor**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

**Comment by guardrex** [comment]
> what kind of services need to be registered over on the Server, as opposed to the client

That's called out by the code block after the line ...

> In the server app's `Startup.ConfigureServices`, register the following additional services:

WRT ...

> Saying "Common Services" is not at all enlightening nor helpful top that end. Can you explain what kinds of services need to be placed here, vs. which should only be registered on the client?

... It's whatever services that your app needs to use in both Razor components and in the server-side app. We can't say what they are ... it depends on your app. We show the minimum services AFAIK. This code came over from the PU for inclusion in the docs. I don't recall running it in a test app. This topic is coming up for a UE ("user experience") pass (i.e., a topic overhaul) on https://github.com/dotnet/AspNetCore.Docs/issues/19286, and I've added a cross-link to this issue there so I won't forget to work on this section further.

## 🐞 

I think that there's a **_bug_** 🐞 in the works here. The line `services.Add...;` should be in the `ConfigureCommonServices` method. I'll take care of that.

**Comment by guardrex** [comment]
I found the original content ...

https://github.com/dotnet/aspnetcore/issues/15253#issuecomment-604581439

"Common services" aren't described in detail, and I took it to mean any service that your app needs to use both via Razor components and via the server-side app.

@javiercn TL;DR ... In your remarks for supporting auth with prerendering, you sent over ...

> On your Client/Program.cs factor common service registrations into a separate method.

... and we have the code to show setting up a `ConfigureCommonServices` method.

"Common services" are just services that both apps need for developer purposes, right? Thare aren't "common" framework services that are placed in the `ConfigureCommonServices` method, correct?
